app/LLM_extractions/extractions.py

```diff
@@ -74,31 +76,54 @@ def extract_information(context: str) -> Any:
         },
     )
 
-    # Create extraction chain
-    chain = prompt | llm | parser
+    retries = 3
+    current_attempt = 0
 
-    # Invoke chain with provided context
-    response = chain.invoke({"context": context})
+    while current_attempt < retries:
```

Comment:
I applied this because sometimes when a query is passed to the user it throws an error but if I pass it again it gives the correct output..hence i applied this. I will try to see if there is any alternative to this.

Reply:
I see, I was hoping Lang Chain had a "native" solution for this.

Reply:
I could not find any solution that Langchain had for this, instead I have now used the `tenacity` library to handle retries in a cleaner way.